### ISSUE TITLE
Use Maria-specific logic with Mysql proper?

### ISSUE THREAD

**Issue by pmjones** [atlasphp/Atlas.Info#12]
Hey @far-blue would it make sense to remove the $maria flag from your recent work, and apply it to all Mysql databases? E.g., it seems like ...

```php
        if (
            $this->maria
            && $column['notnull'] == 0
            && $column['default'] === 'NULL'
        ) {
            $column['default'] = null;
        }

        if (
            $this->maria
            && (in_array($column['type'], ['char', 'varchar', 'text']))
            && $column['default'] === '\'\''
        ) {
            $column['default'] = '';
        }
```

... would be applicable in MySQL as well.

**Comment by far-blue** [comment]
I've honestly not tested. I had assumed (maybe incorrectly) that the code originally worked for MySQL and that MariaDB has changed something over time to cause the divergence of behaviour.

I'll try and find a moment to experiment with a MySQL version of our MariaDB database and see if there are differences.

**Comment by pmjones** [comment]
> I'll try and find a moment to experiment with a MySQL version of our MariaDB database and see if there are differences.

That would be wonderful; thank you!

**Comment by far-blue** [comment]
Ok, I've done some experimenting and found the following.

Firstly, the fixes for MariaDB have zero impact on MySQL (v5.7 and 8.0). MySQL doesn't have the same issues with `'NULL'` and `'\'\''` being reported in default values in the same way MariaDB does. However, I'd prob. still lean towards keeping the mariadb check in place because technically without it if someone stored the string `'NULL'` in a mysql db as the default for a field it could trigger the mariadb fix code and unexpectedly change it to `null`, as an example.

While experimenting I did also notice another possible issue which is the same for MariaDB and MySQL.

Even when `NOT NULL` is set for a text, char or text field, if no `DEFAULT` is provided the generated classes still set the default value to `null`. As this hasn't been raised before I suspect the DB just handles it by converting `null` to `''` if you try to insert or update and I guess we can't do much about 'fixing' this because existing php may rely on a field having a value of `null` rather than `''`.

Any thoughts from anyone?

**Comment by pmjones** [comment]
I think we're good -- until further notice, this is all OK.
